pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -33,9 +33,9 @@ in
 , profilingDetail ? "exported-functions"
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? false
-, enableSharedLibraries ? (ghc.enableShared or false)
+, enableSharedLibraries ? !stdenv.hostPlatform.isStatic && (ghc.enableShared or false)
 , enableDeadCodeElimination ? (!stdenv.isDarwin)  # TODO: use -dead_strip for darwin
-, enableStaticLibraries ? !(stdenv.hostPlatform.isWindows or stdenv.hostPlatform.isWasm)
```

Comment:
Is this change necessary?  I think [`!(a or b) == !a and !b`](https://en.wikipedia.org/wiki/De_Morgan%27s_laws).

Although it would be really nice to have a comment above this line explaining why static libraries are disabled when the host platform is windows or wasm.

Reply:
Yes I thought i was going to change it more but then I didn't. I'll put back.